core/image/etc/etc2_compressor.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright (C) 2021 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <assert.h>
+#include <stdlib.h>
+#include <string.h>
+#include "etc2.h"
+#include "third_party/etc2comp/EtcLib/Etc/EtcImage.h"
+
+static_assert(sizeof(etc_error) >= sizeof(Etc::Image::EncodingStatus),
+              "astc_error should superset of astcenc_error");
+
+namespace {
+const int64_t ETC_ALLOCATION_FAILED = -1;
+const uint32_t MIN_JOBS = 8;
+const uint32_t MAX_JOBS = 1024;
+const float EFFORT = 10.0f;
```

Comment:
So, this is currently hardcoded but I will make it set by the user when I integrate the feature completely. For now, It is intentionally low value because compressing ETC takes too much time and causes timeout in compression test. We don't need to test image quality or anything, we just need to ensure that the compression is actually takes place. Therefore I kept it low. I just kept the simplest implementation for now, until I do the integration work which is soon.

Reply:
Makes sense.

That's the kind of info you probably want to put in the commit message (to avoid reviewers wondering why this is currently hardcoded :slightly_smiling_face: ).